src/mds/MDSMap.h

```diff
@@ -150,6 +150,21 @@ class MDSMap {
     void encode_unversioned(bufferlist& bl) const;
   };
 
+  struct mds_rank_node {
+      mds_rank_t rank;
+      uint32_t hash;
+      vector<inodeno_t> inode_nos;
+    };
+
+  std::vector<mds_rank_node> mds_rank_nodes = {{0, rjhash32(0)}};
+  void add_rank_node_to_consistent_hash_ring(mds_rank_t rank);
+  mds_rank_node get_rank_node_from_ino_in_consistent_hash_ring(inodeno_t ino) const; //Get the rank that handles the key i.e inode number
+  mds_rank_t put_ino_in_consistent_hash_ring(inodeno_t ino); //Assign the key i.e inode number to a rank in the consistent hash ring
+  vector<inodeno_t> get_inos_from_rank_in_consistent_hash_ring(mds_rank_t rank) const;
+  mds_rank_t get_successor_of_rank_in_consistent_hash_ring(mds_rank_t rank) const;
+  void remove_rank_node_from_consistent_hash_ring(mds_rank_t rank);
```

Comment:
I have to pause my review at this point but I was surprised to find these methods here. I would expect most of this logic to exist in `MDCache`. Don't move anything yet as I'm not sure about this data structure either.

Let's talk about this during standup.